libs/core/algorithms/tests/unit/algorithms/adjacentdifference_tests.hpp

```diff
@@ -83,8 +84,14 @@ void test_adjacent_difference_sender(Policy l, ExPolicy&& policy)
     using scheduler_t = ex::thread_pool_policy_scheduler<Policy>;
 
     auto exec = ex::explicit_scheduler_executor(scheduler_t(l));
+#ifdef HPX_HAVE_STDEXEC
+    auto result =
+        tt::sync_wait(ex::just(std::begin(c), std::end(c), std::begin(d)) |
+            hpx::adjacent_difference(policy.on(exec)));
+#else
     auto result = ex::just(std::begin(c), std::end(c), std::begin(d)) |
         hpx::adjacent_difference(policy.on(exec)) | tt::sync_wait();
+#endif
```

Comment:
that duplication is present in many of the tests i have changed so it might make sense to do that there too?
e.g. https://github.com/STEllAR-GROUP/hpx/pull/6431/files#diff-b11a40f80e11363af9d5469224fdb4e94487ee5ff99c30c7d4219a5ab40c85d8